src/main/java/org/fogbowcloud/saps/engine/core/archiver/storage/swift/SwiftPermanentStorage.java

```diff
@@ -208,26 +227,106 @@ public boolean delete(SapsImage task) {
         LOGGER.debug("Deleting files from task [" + taskId + "] in Swift [" + containerName + "]");
 
         String swiftExports = (task.getState() == ImageTaskState.FAILED && this.debugMode)
-                ? properties.getProperty(SapsPropertiesConstants.PERMANENT_STORAGE_DEBUG_TASKS_DIR)
-                : properties.getProperty(SapsPropertiesConstants.PERMANENT_STORAGE_TASKS_DIR);
-
-        String prefix = swiftExports + File.separator + taskId;
+                ? debugTasksDirName
+                : tasksDirName;
 
-        List<String> fileNames = swiftAPIClient.listFilesWithPrefix(containerName, prefix);
+        String dir = swiftExports + File.separator + taskId;
 
-        LOGGER.info("Files List: " + fileNames);
+        try {
+            List<String> fileNames = swiftAPIClient.listFiles(containerName, dir);
+            LOGGER.info("Files List: " + fileNames);
 
-        for (String file : fileNames) {
-            LOGGER.debug("Trying to delete file " + file + " from " + containerName);
-            swiftAPIClient.deleteFile(containerName, file);
+            for (String file : fileNames) {
+                LOGGER.debug("Trying to delete file " + file + " from " + containerName);
+                swiftAPIClient.deleteFile(containerName, file);
+            }
+        } catch (IOException e) {
+            LOGGER.error("Error while list files of path [" + dir + "] from Object Storage", e);
+            return false;
         }
 
         return true;
     }
 
     @Override
-    public List<AccessLink> generateAccessLinks(String taskId) throws TaskNotFoundException, IOException {
-        return null;
+    public List<AccessLink> generateAccessLinks(String taskId)
+        throws TaskNotFoundException, IOException {
+        List<String> files = this.listFiles(taskId);
+        List<AccessLink> filesLinks = this.generateLinks(files);
+        return filesLinks;
+    }
+
+    private List<String> listFiles(String taskId) throws IOException, TaskNotFoundException {
+        List<String> files = new ArrayList<>();
+
+        String[] dirs = {INPUTDOWNLOADING_DIR, PREPROCESSING_DIR, PROCESSING_DIR};
+
+        if(!this.swiftAPIClient.existsTask(this.containerName, this.tasksDirName, taskId)) {
+            throw new TaskNotFoundException("Task [" + taskId + "] was not found in directory [" + this.tasksDirName + "] of container [" + this.containerName + "]");
+        }
+
+        for (String dir : dirs) {
+            String dirPath = String.format(SWIFT_TASK_STAGE_DIR_PATTERN, this.tasksDirName, taskId, dir);
+            try {
```

Comment:
avoid this nested calls (line 271): it makes debugging harder.

Reply:
Done.